File: src/main/java/main/data/Individual.java
package main.data;

import javax.persistence.*;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Entity
public class Individual implements Comparable<Individual> {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Min(-80)
    @Max(240)
    private int score;

    @NotNull
    private String name;

    @Enumerated(EnumType.STRING)
    @NotNull
    private Division division;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "team_id")
    public Team team;

    private Individual() {
    }

    Individual(String name, Team team, Division division) {
        this.name = name;
        this.team = team;
        this.division = division;
    }

    public Division getDivision() {
        return division;
    }

    void setDivision(Division division) {
        this.division = division;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public Team getTeam() {
        return team;
    }

    void setTeam(Team team) {
        this.team = team;
    }

    @Override
    public int compareTo(Individual o) {
        if(o.score != score) {
            return Integer.compare(o.score, score);
        }
        return name.compareTo(o.name);
    }
}

File: src/main/kotlin/main/data/Individual.kt
package main.data

import javax.persistence.*
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import java.time.LocalDateTime

@Entity
class Individual internal constructor(@field:NotNull
                                      var name: String, team: Team, division: Division) : Comparable<Individual> {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Int = 0
    @Min(-80)
    @Max(240)
    var score: Int = 0

    @Enumerated(EnumType.STRING)
    @NotNull
    var division: Division? = null
        internal set

    @NotNull
    @ManyToOne
    @JoinColumn(name = "team_id")
    var team: Team
        internal set

    @Version
    val timestamp: LocalDateTime? = null

    init {
        this.team = team
        this.division = division
    }

    override fun compareTo(other: Individual): Int {
        return if (other.score != score) {
            other.score.compareTo(score)
        } else name.compareTo(other.name)
    }
}
